test/e2e/helpers_test.go

```
@@ -0,0 +1,323 @@
+	assertDoesNotExist(types.NamespacedName{
+		Name:      hzName + "-0",
+		Namespace: hzNamespace,
+	}, &corev1.Pod{})
+
+	By("Waiting for Hazelcast CR to be removed", func() {
+		Eventually(func() error {
+			h := &hazelcastcomv1alpha1.Hazelcast{}
+			return k8sClient.Get(context.Background(), types.NamespacedName{
+				Name:      hzName,
+				Namespace: hzNamespace,
+			}, h)
+		}, timeout, interval).ShouldNot(Succeed())
+	})
+}
+func DeletePod(podName string, gracePeriod int64) {
+	log.Printf("Deleting POD with name '%s'", podName)
+	deleteOptions := metav1.DeleteOptions{
+		GracePeriodSeconds: &gracePeriod,
+	}
+	err := GetClientSet().CoreV1().Pods(hzNamespace).Delete(context.Background(), podName, deleteOptions)
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func GetHzClient(ctx context.Context, unisocket bool) (*hzClient.Client, error) {
+	var lookupKey = types.NamespacedName{
+		Name:      hzName,
+		Namespace: hzNamespace,
+	}
+	s := &corev1.Service{}
+	Eventually(func() bool {
+		err := k8sClient.Get(context.Background(), lookupKey, s)
+		Expect(err).ToNot(HaveOccurred())
+		return len(s.Status.LoadBalancer.Ingress) > 0
+	}, timeout, interval).Should(BeTrue())
+	addr := s.Status.LoadBalancer.Ingress[0].IP
+	if addr == "" {
+		addr = s.Status.LoadBalancer.Ingress[0].Hostname
+	}
+	Expect(addr).Should(Not(BeEmpty()))
+	By("connecting Hazelcast client")
+	config := hzClient.Config{}
+	config.Cluster.Network.SetAddresses(fmt.Sprintf("%s:5701", addr))
+	config.Cluster.Unisocket = unisocket
+	config.Cluster.Discovery.UsePublicIP = true
+	client, err := hzClient.StartNewClientWithConfig(ctx, config)
+	Expect(err).ToNot(HaveOccurred())
+	return client, err
+}
+
+func GetClientSet() *kubernetes.Clientset {
+	kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(clientcmd.NewDefaultClientConfigLoadingRules(), &clientcmd.ConfigOverrides{})
+	restConfig, err := kubeConfig.ClientConfig()
+	if err != nil {
+		log.Fatal(err)
```

I assume Failing or Expecting nil here instead of logging would be better. WDYT?

---

`log.Fatal` is actually failing the proccess

---

Ah, didn't know that. :+1:

---

Yeap, that's a bit counterintuitive. I only know this as we already discussed it with @cheels once 😃

---

Got it :D